packages/dev/core/src/Engines/engine.ts

```diff
@@ -697,12 +697,10 @@ export class Engine extends ThinEngine {
             }
         };
 
-        if (IsWindowObjectExist()) {
-            const hostWindow = this.getHostWindow();
-            if (hostWindow) {
-                hostWindow.addEventListener("blur", this._onBlur);
-                hostWindow.addEventListener("focus", this._onFocus);
-            }
+        const hostWindow = this.getHostWindow(); // it calls IsWindowObjectExist()
+        if (hostWindow && typeof hostWindow.addEventListener === 'function') {

```

Comment:
make sure `addEventListener` exists and is function.

Note that in my case there is no exception thrown here, because we don't go through this code from the `NullEngine` constructor. I did it for consistency and symetry with what I changed for the removeEventListener